pages/lib/template_graphs.py

```diff
@@ -404,31 +412,103 @@ def heatmap_with_filter(
         # Set maximum and minimum according to data
         data_max, data_min = get_max_min_value(df[var])
         range_z = [data_min, data_max]
-    fig = go.Figure(
-        data=go.Heatmap(
-            y=df[Variables.HOUR.col_name]
-            - 0.5,  # Offset by 0.5 to center the hour labels
-            x=df[Variables.UTC_TIME.col_name].dt.date,
-            z=df[var],
-            colorscale=var_color,
-            zmin=range_z[0],
-            zmax=range_z[1],
-            customdata=np.stack(
-                (df[Variables.MONTH_NAMES.col_name], df[Variables.DAY.col_name]),
-                axis=-1,
-            ),
-            hovertemplate=(
-                "<b>"
-                + var
-                + ": %{z:.2f} "
-                + var_unit
-                + "</b><br>Month: %{customdata[0]}<br>Day: %{customdata[1]}<br>Hour:"
-                " %{y}:00<br>"
-            ),
-            name="",
-            colorbar=dict(title=var_unit),
+    fig = go.Figure()
+
+    has_filter_marker = "_is_filtered" in df.columns
```

Comment:
possible code duplication with onther function which has very similar code. We should avoid code duplications